src/API/Client.php

```diff
@@ -0,0 +1,203 @@
+<?php
+declare(strict_types=1);
+
+namespace RocketCDN\API;
+
+use RocketCDN\Options\Options;
+
+class Client {
+	const ROCKETCDN_API = 'https://rocketcdn.me/api/';
+
+	/**
+	 * Options instance
+	 *
+	 * @var Options
+	 */
+	private $options;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param Options $options Options instance.
+	 */
+	public function __construct( Options $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Gets the customer data associated with the API key
+	 *
+	 * @param string $api_key the API key to authenticate the request.
+	 *
+	 * @return array
+	 */
+	public function get_customer_data( $api_key = '' ): array {
+		$cache = get_transient( 'rochetcdn_customer_data' );
+
+		if ( false !== $cache ) {
+			return $cache;
+		}
+
+		$result = $this->get_raw_customer_data( $api_key );
+
+		if ( empty( $result ) ) {
+			return [];
+		}
+
+		set_transient( 'rocketcdn_customer_data', $result, 2 * DAY_IN_SECONDS );
+
+		return $result;
+	}
+
+	/**
+	 * Gets the customer data from the API
+	 *
+	 * @param string $api_key the API key to authenticate the request.
+	 *
+	 * @return array
+	 */
+	private function get_raw_customer_data( $api_key = '' ): array {
+
+		if ( empty( $api_key ) ) {
+			return [];
+		}
+
+		$response = wp_remote_get(
+			self::ROCKETCDN_API . 'customer/me',
```

Comment:
Does it make any sense here to set up some kind of filter for using a different (testing) api? Or perhaps check for a constant set in wp-config.php?

Reply:
I'm always conflicted about filters or constants that are for testing purpose only, and not for end-users. If anything, I would rather use a filter than a constant.

What's your opinion on this?

Reply:
I feel ya.
On one hand it's not essential at this stage. We can always edit the `const` during e2e testing, if we're doing it manually. We will need a way to modify it programmatically if ever we get to the point of automating e2e (which may or may not ever happen).
On the other hand, a filter could be a benefit to end users. Use case: something changes on the CDN itself on short notice and we have to switch the URL temporarily. Support could simply indicate to use the filter and customers could avoid waiting for a release.

Reply:
Would you filter the whole URL, or the base API URL only? As the ROCKETCDN_API is also used for the purge cache function

Reply:
just the base. Why? because presumably any staging or temporary environment would match the path structure of the production one. So swapping the base only will make everything else track the same on other environments without having to worry about the rest of the pathing.

Reply:
Added filter in https://github.com/wp-media/rocketcdn-plugin/pull/8/commits/da378f1df4b04439b42583840fd72979e1462933

Reply:
That totally makes sense!!! :) 👍